### ISSUE TITLE
Insert sort

### ISSUE THREAD

**Issue by kwarkGorny** [skypjack/entt#255]
Hello, I am working on little game and I faceing problem with Layering of sprites. Lets assume we have SpriteComponent and we want to keep them sorted by LayerComponent (or by some layer value in SpriteComponent) is there a way to insert sort components so instead of doing this 
'''
registry.assign<SpriteComponent>(...);
registry.sort<SpriteComponent>(...);
'''
I can do something like
'''
registry.assing_sorted<SpriteComponent>(entityID, compare, ...)
'''
or just set it once somewhere at the start like
'''
registry.insert_sort<SpriteComponent>(compare, algorithm, ...)
'''
and after this setup each assign of component would be insert sorted

**Comment by Kerndog73** [comment]
Worst-case scenario is that you have to sort once every frame. You only need to sort at the end of the frame when something changed during that frame. You could just keep a dirty flag for that. Going beyond that kind of sounds like premature optimisation too me. 🤔

I think I remember seeing insertion sort or bubble sort somewhere but it might have been deleted.

**Comment by skypjack** [comment]
Insertion sort is an algorithm and [it's still available](https://github.com/skypjack/entt/blob/master/src/entt/core/algorithm.hpp#L43). In general, it's a better choice in most cases when sorting views or groups.  
On the other side, `assign_sorted` doesn't look like a good idea to me. A sparse set isn't a plain vector, sorting them in EnTT relies on a trick to make it faster than usual. `assign_sorted` would require to partially rebuild the sparse array after each insertion instead, other than to memcpy parts of the internal arrays back and forth, when probably all what you want is to swap two entities and their components in most cases.

That said, if you know in advance how many layers you have, you can avoid all of this using a more data oriented approach.

**Comment by kwarkGorny** [comment]
@skypjack  No, but thanks for replays and suggestions I just end up  making something like this :
`
registry.group<Background>(entt::get<sf::Sprite, TransformComp>).each(...);
registry.group<Gameplay>(entt::get<sf::Sprite, TransformComp>).each(...);
registry.group<Gui>(entt::get<sf::Sprite, TransformComp>).each(...);
`
and its good enough for my project

**Comment by skypjack** [comment]
I see. I want also to invite you to join the [gitter channel](https://gitter.im/skypjack/entt) if you've any other question about `EnTT` or component based design in general. :+1:
